src/material/form-field/form-field.ts

```diff
@@ -356,8 +347,13 @@ export class MatFormField extends _MatFormFieldMixinBase
 
     if (this._dir) {
       this._dir.change.pipe(takeUntil(this._destroyed)).subscribe(() => {
-        this.updateOutlineGap();
-        this._previousDirection = this._dir.value;
+        if (typeof requestAnimationFrame === 'function') {
+          this._ngZone.runOutsideAngular(() => {
```

Comment:
Shouldn't this whole `if-else` be wrapped in `runOutsideAngular`? Currently it looks like it runs in the zone if the browser doesn't support `requestAnimationFrame` and outside if it does

Reply:
`updateOutlineGap` doesn't do any zone-ey things, we only need to run `requestAnimationFrame` on the outside because it can trigger change detection otherwise.